src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -152,11 +153,9 @@ public void Start()
                     return;
                 }
 
-                //- It is raised when the underlying connection closes because of our close operation
-                var messagingException = exceptionReceivedEventArgs.Exception as MessagingException;
-                if (messagingException != null && messagingException.IsTransient)
+                if (exceptionReceivedEventArgs.Exception.IsTransientException(exceptionReceivedEventArgs.Action))
```

Comment:
With the change to how we'll evaluate if exception is transient or not, it will be less verbose.
I'm not sure `exceptionReceivedEventArgs.IsTransient()` would read right as it's the exception that should be checked for being transient, not the arguments.